### ISSUE TITLE
CPU usage issue

### ISSUE THREAD

**Issue by npbdevelop** [microsoft/vscode#215747]
Type: <b>Performance Issue</b>

Ever since 2 days ago VS Code has (seemingly randomly) started consuming an enormous amount of my CPU resources. It happens in nearly all contexts, and I've noticed it gets exponentially worse as I increase the size of the window. I'll list some details, then what I've tried.
- Primarily affects typing and hovering cursor over objects in window, so editing code and selecting options is very slow
- Does not seem to affect dropdown menu options, eg selecting file--preferences is lightning quick, but the moment I open the settings page, performance takes a nosedive
-  Process explorer reports practically 0 CPU usage but top in command line tells me otherwise, often 100% or more usage

- I've reinstalled VS code multiple times, with snap and with store
- I've deleted all settings in json files
- I've updated and cleaned my system files
- I've investigated all other applications on my system, including Qt creator, which performs exactly as expected
- I've opened simply blank text files and it still occurs the moment I start typing or clicking
- I've deleted and disabled all extensions

I do not recall a specific change I made, though I do know VS Code updated right before it happened.

VS Code version: Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:43.666Z)
OS version: Linux x64 6.8.0-35-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5600X 6-Core Processor (12 x 4641)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 2, 1|
|Memory (System)|15.52GB (11.60GB free)|
|Process Argv|--force-user-env --no-sandbox --use-gl=angle --use-angle=swiftshader --crash-reporter-id ee063c39-d7e2-4d8b-8896-471014ae3e50|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   159	 18873	code main
    0	    32	 18875	   zygote
    0	    32	 18876	   zygote
    0	   222	 18953	window [1] (Welcome - Visual Studio Code)
    0	    95	 19084	window [2] (Issue Reporter)
    0	    64	 18910	   utility-network-service
    0	   111	 18991	extensionHost [1]
    0	    79	 19041	     /snap/code/161/usr/share/code/code /snap/code/161/usr/share/code/resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=18991
    0	    95	 19001	shared-process
    0	     0	 19096	     /bin/sh -c /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 19097	       /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	    79	 19002	fileWatcher [1]
```

</details>
<details>
<summary>Workspace Info</summary>

```
;
```

</details>
Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyonecf:30548226
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelt:31048053
dsvsc021:30996838
945dj816:31013170
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupytercf:31046870

```

</details>

**Comment by shelvesdragon** [comment]
This is likely caused by the temporary fix in version 1.90.1 on the snap package that switched the OpenGL backend to software emulation in order to prevent the crashes reported with version 1.90.0 in #212494 . If it is related, your options for now are mentioned in this comment: https://github.com/microsoft/vscode/issues/212494#issuecomment-2168072242

Snap and the Ubuntu store install the same snap package and revision, which is v1.90.1. Ubuntu store just does so with more GUI. Given that you uninstalled and reinstalled the snap package, option 2 a) `snap revert code --revision 159` likely won't work for you. You'd have to remove VSCode:
```bash
sudo snap remove code
```
And reinstall revision 159 with:
```bash
sudo snap install code --revision 159 --classic
```
Note: This will keep revision 159 only until the next release and will not remove user data, i. e. opened workspace, extensions etc.

Hope this helps!

**Comment by deepak1556** [comment]
Merging to https://github.com/microsoft/vscode/issues/212494, please follow the issue for updates.
